tst/regression/test_suites/restart/restart.py

```diff
@@ -22,41 +22,43 @@
 import sys
 import os
 import utils.test_case
+
+
 """ To prevent littering up imported folders with .pyc files or __pycache_ folder"""
 sys.dont_write_bytecode = True
 
 class TestCase(utils.test_case.TestCaseAbs):
-    def Prepare(self, parameters, step):
-            
-        # enable coverage testing on pass where restart
-        # files are both read and written
-        parameters.coverage_status = "both"
-        
-        if step == 1:
-            parameters.driver_cmd_line_args = ['parthenon/job/problem_id=gold']
-        else:
-            parameters.driver_cmd_line_args = [
-                '-r',
-                'gold.out0.00001.rhdf',
-                'parthenon/job/problem_id=silver'
-            ]
-            
-        return parameters
-
-    def Analyse(self, parameters):
-        # spotcheck one variable
-        goldFile = 'gold.out0.00002.rhdf'
-        silverFile = 'silver.out0.00002.rhdf'
-
-        gold = h5py.File(goldFile,'r')
-        silver = h5py.File(silverFile,'r')
-
-        varName = "/advected"
-        goldData = gold[varName][:].flatten()
-        silverData = gold[varName][:].flatten()
-
-        # spot check on one variable
-        maxdiff = max(abs(goldData-silverData))
-        print('Variable: %s, diff=%g, N=%d'%(varName,maxdiff,len(goldData)))
-
-        return (maxdiff == 0.0)
+  def Prepare(self, parameters, step):
+
+    # enable coverage testing on pass where restart
+    # files are both read and written
+    parameters.coverage_status = "both"
+
+    if step == 1:
+      parameters.driver_cmd_line_args = ['parthenon/job/problem_id=gold']
+    else:
+      parameters.driver_cmd_line_args = [
+        '-r',
+        'gold.out0.00001.rhdf',
+        'parthenon/job/problem_id=silver'
+      ]
+
+    return parameters
+
+  def Analyse(self, parameters):
+    # spotcheck one variable
+    goldFile = 'gold.out0.00002.rhdf'
+    silverFile = 'silver.out0.00002.rhdf'
+
+    varName = "advected"
+    with h5py.File(goldFile,'r') as gold, h5py.File(silverFile,'r') as silver:
+      goldData = np.zeros(gold[varName].shape, dtype=np.single)
```

Comment:
Our default config is double precision in Parthenon so we also write/read double precision outputs for the restart files.
Potentially, that may require lowering the diff threshold below.

```suggestion
      goldData = np.zeros(gold[varName].shape, dtype=np.float64)
```